web/src/components/Agency/AgencyForm/AgencyForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { Button } from 'react-bootstrap'
+import { useForm, UseFormReturn } from 'react-hook-form'
+import type { EditAgencyById, UpdateAgencyInput } from 'types/graphql'
+
+import {
+  Form,
+  FormError,
+  FieldError,
+  Label,
+  TextField,
+  Submit,
+} from '@redwoodjs/forms'
+import type { RWGqlError } from '@redwoodjs/forms'
+
+type FormAgency = NonNullable<EditAgencyById['agency']>
+
+interface AgencyFormProps {
+  agency?: EditAgencyById['agency']
+  onSave: (data: UpdateAgencyInput, id?: FormAgency['id']) => void
+  error: RWGqlError
+  loading: boolean
+}
+
+const AgencyForm = (props: AgencyFormProps) => {
+  const { agency, onSave, error, loading } = props
+  const formMethods: UseFormReturn<FormAgency> = useForm<FormAgency>()
+  const hasErrors = Object.keys(formMethods.formState.errors).length > 0
+
+  // Resets the form to the previous values when editing the existing agency
+  // Clears out the form when creating a new agency
+  const onReset = () => {
+    formMethods.reset()
+  }
+
+  const onSubmit = (data: FormAgency) => {
+    onSave(data, props?.agency?.id)
+  }
+
+  return (
+    <Form<FormAgency>
+      onSubmit={onSubmit}
+      formMethods={formMethods}
+      error={error}
+      className={hasErrors ? 'was-validated' : ''}
+    >
+      {agency && (
+        <div className="row">
+          <Label name="id" className="form-label col-sm-2 col-form-label">
+            Agency Code
```

Comment:
Is Agency Code supposed to map to the agency.id? I don't think we need to hold up this PR for this but it is probably worth at least noting this explicitly.